src/Nethermind/Nethermind.Blockchain/Receipts/PersistentReceiptStorage.cs

```diff
@@ -225,6 +225,8 @@ public void Insert(Block block, params TxReceipt[] txReceipts)
             RlpBehaviors behaviors = spec.IsEip658Enabled ? RlpBehaviors.Eip658Receipts | RlpBehaviors.Storage : RlpBehaviors.Storage;
             _blocksDb.Set(block.Hash, StorageDecoder.Encode(txReceipts, behaviors).Bytes);
             
+            ReceiptsInserted?.Invoke(this, new ReceiptsEventArgs(block.Header, txReceipts));
```

Comment:
calling Receipts inserted before setting fields like TxHash feels like something that may lead to unexpected behaviour from event handlers

Reply:
Move after the loop